src/index.cpp

```diff
@@ -1708,7 +1708,10 @@ static void writeNamespaceTree(const NamespaceLinkedRefMap &nsLinkedMap,FTVHelp
 {
   for (const auto &nd : nsLinkedMap)
   {
-    writeNamespaceTreeElement(nd,ftv,rootOnly,addToIndex);
+    if (nd->isLinkableInProject())
```

Comment:
This will also hide documented namespaces imported via tag files. Isn't the real problem that undocumented namespaces appeared in the tag file?

Reply:
I don't think that the problem is that undocumented namespaces appear in the tag file as undocumented classes appear also in the tag file and the documented and undocumented classes don't appear in the tree of the project that uses this tag file.
I created a small example to illustrate this: [example.tar.gz](https://github.com/doxygen/doxygen/files/8495000/example.tar.gz)

And in the output we see:

![image](https://user-images.githubusercontent.com/5223533/163549540-a8c27881-e3b9-40e2-b1f6-92f9086be606.png)

So no classes from project 1 but the namespaces from project 1 are shown.

I think logic of the proposed implementation is correct.

Reply:
> This will also hide documented namespaces imported via tag files. Isn't the real problem that undocumented namespaces appeared in the tag file?

I would like to clarify that in the original issue when I stated "undocumented external namespaces..." I meant external namespaces that were **not documented in my project**. The namespaces in my example picture are indeed documented in the original QDoc project that generated those tags.

I reported this as a bug as I was under the impression that the purpose of importing tags is so that a user's documentation can easily reference/link to external documentation (i.e. its dependencies), and not much more.

In my opinion, the primary listing pages (i.e. Namespaces, Classes, Files, etc.) should only list members that are part of the project in question, and not those that are from external projects, unless for whatever reason you explicitly added documentation for them in your own. They way the Namespaces page appears now gives the false impression that my library defines those extra namespaces, when in realty it doesn't define or add to them in any way.

As Albert pointed out, the Classes listing already behaves the "ideal" way (as does Files). This, along with the fact that the external namespaces that do appear have no brief descriptions, no pages associated with them, and aren't even linked to the external documentation that they are documented in, makes this seem like much more of a bug than intentional.

At least, that's how it seems to me.